src/mlpack/methods/decision_tree/decision_tree_impl.hpp

```diff
@@ -1143,12 +1143,15 @@ void DecisionTree<FitnessFunction,
   }
 
   // Serialize the children first.
-  ar & BOOST_SERIALIZATION_NVP(children);
+  for (size_t i = 0; i < children.size(); ++i)
+  {
+    ar & CEREAL_POINTER(children.at(i));
+  }
```

Comment:
Yes, I wanted to ask about that, but since it worked when I serialized one by one I moved on.
I agree that we need to create a wrapper, just to register the number on the children

Reply:
:+1: yeah, it should not be too hard to make a wrapper.